### ISSUE TITLE
[BUG] Microsoft.Rest.Azure.CloudException When Trying To Iterate Over a Large Number of Container Groups

### ISSUE THREAD

**Issue by Rhipeus** [Azure/azure-libraries-for-net#919]
**Describe the bug**
When iterating over a large number of container groups, at the 581st iteration, there is always a "Client certificate was not provided" exception.

***Exception or Stack Trace***
Microsoft.Rest.Azure.CloudException: A client certificate was not provided.

**To Reproduce**
Create an Azure subscription with lots of container groups.  Try to iterate over all of them.

***Code Snippet***
```
using System;
using System.Threading.Tasks;
using Microsoft.Azure.Management.Fluent;
using Microsoft.Azure.Management.ResourceManager.Fluent;

namespace TestGetContainerInstances
{
    class Program
    {
        static async Task Main(string[] args)
        {
            var i = 1;

            foreach (var item in GetAzureContext().ContainerGroups.List()) //Errors when i==581
            {
                Console.WriteLine($"{i++} {item.Name}");
            }
        }

        private static IAzure GetAzureContext()
        {
            IAzure azure = null;

            var credentials = SdkContext.AzureCredentialsFactory.FromServicePrincipal(
                "<redacted client id>",
                "<redacted secret>",
                "<redacted tenant>",
                AzureEnvironment.AzureGlobalCloud);

            azure = Azure.Authenticate(credentials).WithDefaultSubscription();

            return azure;
        }
    }
}
```

**Expected behavior**
Be able to iterate over all container groups.

**Setup (please complete the following information):**
 - OS: Windows 10 and Azure WebJob
 - IDE : Visual Studio 2019
 - Microsoft.Azure.Management.Fluent v1.28.1

**Additional context**
The Azure subscription has been set to allow up to 1000 concurrent container groups instead of the default 100.

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Bug Description Added
- [X] Repro Steps Added
- [X] Setup information Added

**Comment by yaohaizh** [comment]
@Rhipeus thanks for reporting the issue. We will look into the issue.

**Comment by Rhipeus** [comment]
I added .NET version to the original report.  

Also, here is some additional information from my digging.  I tried adding/removing some container groups in Azure so that #581 was not the same container group.  It continued to error on #581 even though it was a different container group.

Below 580 container groups, everything worked as expected.

**Comment by ChenTanyi** [comment]
@Rhipeus After the subscription being set to 1000 concurrent container groups, I have tested it. But I cannot reproduce the problem. As my result showing, I can iterate all 1000 container groups.
![image](https://user-images.githubusercontent.com/15213781/74013514-92e4a900-49c7-11ea-9c21-6e147c9b7bdf.png)

I'm not sure is there anything related to your runtime environment, since actually when you list over all container groups, it would use the same token, which could not fail in the middle of all requests. The only difference is that my .NET framework is 4.8 (but I think it doesn't matter).

Could you give me more information about your http requests, the 580th request (success) and the 581st request (failure), especially their headers for token?
> You can install a [Fiddler](https://www.telerik.com/fiddler) or [Charles](https://www.charlesproxy.com/), install their root certificate and open the system proxy, then you can catch all http requests sent by visual studio.

**Comment by Rhipeus** [comment]
I would agree that it shouldn't have failed in the middle of all requests but something made it so that it could fail in the middle of all requests.

I'm not sure that I can provide much more information at this time.  This issue occurred and broke our production system.  We have configured our system to only use 500 of our 1000 available slots to avoid this problem.  We are concerned that increasing the number again will break our production system.  We do not have any other systems with more than the default 100 available docker container groups available to test with.

In order to get you this information, we will have to schedule maintenance time for our production system.  Scheduling this will be at least 2 weeks out at this point.  Because it took 2 months to get any response, I've actually been hopeful that something has already been fixed in the meantime.

As for the runtime environment, this issue originally occurred running as an Azure WebJob in an app service.  I was able to recreate it with the code above on my development windows 10 environment.

**Comment by ChenTanyi** [comment]
@Rhipeus Sorry for the late reply, since you said that you changed the container groups but it also raised an exception at 581st, I'd like to test if the 581 is a magic number, which costs lots of time waiting for expanding the container group number.

It seems that you can reproduce it in your subscription for any client environment, I'd prefer to think it was a problem in your azure runtime services. And I found that you exception is "A client certificate was not provided.", but actually you don't need the send any client certificate when you list the container groups. So I think it may be a problem in your container group.

May I ask you some of the question:
* Have you used other azure-tools or azure-sdk to list the container groups?
> Like azure-cli, azure-powershell, java-sdk, python-sdk, or even rest-api
* How does your 1000 container groups be arranged? All in one resource group or the first 580 container groups are in different resource group from the rest?
* Do you know what is the "client certificate" used for? I think it is a certificate used inside container groups?

Meanwhile, since you had dealt with it by reducing the number of container groups, I think you don't need to expand it in order to test the bugs in your production environment, if the 500 container groups meet your production requirements.
